src/ui/pages/Menu/components/Settings/components/ManagePassword/ManagePassword.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  IonCard,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonModal,
+  IonToggle,
+} from "@ionic/react";
+import { useSelector } from "react-redux";
+import { useRef, useState } from "react";
+import { chevronForward } from "ionicons/icons";
+import { i18n } from "../../../../../../../i18n";
+import { VerifyPassword } from "../../../../../../components/VerifyPassword";
+import { VerifyPasscode } from "../../../../../../components/VerifyPasscode";
+import {
+  getStateCache,
+  setToastMsg,
+} from "../../../../../../../store/reducers/stateCache";
+import {
+  Alert as AlertEnable,
+  Alert as AlertDisable,
+} from "../../../../../../components/Alert";
+import { CreatePassword } from "../../../../../CreatePassword";
+import { KeyStoreKeys, SecureStorage } from "../../../../../../../core/storage";
+import { useAppDispatch } from "../../../../../../../store/hooks";
+import { ToastMsgType } from "../../../../../../globals/types";
+
+const ManagePassword = () => {
+  const dispatch = useAppDispatch();
+  const stateCache = useSelector(getStateCache);
+  const userAction = useRef("");
+  const [passwordIsSet, setPasswordIsSet] = useState(
+    stateCache?.authentication.passwordIsSet
+  );
+  const [alertEnableIsOpen, setAlertEnableIsOpen] = useState(false);
+  const [alertDisableIsOpen, setAlertDisableIsOpen] = useState(false);
+  const [verifyPasswordIsOpen, setVerifyPasswordIsOpen] = useState(false);
+  const [verifyPasscodeIsOpen, setVerifyPasscodeIsOpen] = useState(false);
+  const [createPasswordModalIsOpen, setCreatePasswordModalIsOpen] =
+    useState(false);
+
+  const handleToggle = () => {
+    if (passwordIsSet) {
+      userAction.current = "disable";
+      setAlertDisableIsOpen(true);
+    } else {
+      userAction.current = "enable";
+      setAlertEnableIsOpen(true);
+    }
+  };
+
+  const handleClear = () => {
+    setAlertEnableIsOpen(false);
+    setAlertDisableIsOpen(false);
+    userAction.current = "";
+    setCreatePasswordModalIsOpen(false);
+  };
+
+  const onVerify = async () => {
+    if (passwordIsSet && userAction.current === "disable") {
+      await SecureStorage.set(KeyStoreKeys.APP_OP_PASSWORD, "");
```

Comment:
There is a `.delete` function

Reply:
That will delete the key, and we don't want that because it will throw an error later on if we try to set a password, it has to be an empty string.

Reply:
Why does it have to be? I don't think we should be storing an empty string with the enclave. Why can't we keep it unset, and use `.delete` when removing it? (And check for the appropriate error when fetching it to see if it's set or not)